Added new kb article numerictextbox-disable-arrows (#2398)
telerik/blazor-docs: knowledge-base/numerictextbox-disable-arrows.md

---

> title: Disabling NumericTextBox Arrows at Min or Max Value in Blazor

Reviewer: People normally don't search in Google with gerund ("-ing"), so it's better that we don't use this grammar form in titles.

description: Learn how to programmatically disable the increase or decrease arrows of a NumericTextBox in Blazor when the value reaches its minimum or maximum limit.
type: how-to
page_title: How to Disable NumericTextBox Arrows on Min or Max Value in Blazor
slug: numerictextbox-disable-arrows
tags: numerictextbox, blazor, disable arrows, conditional styling
res_type: kb
ticketid: 1665216
---

## Environment

<table>
    <tbody>
	    <tr>
	    	<td>Product</td>
	    	<td>NumericTextBox for Blazor</td>
	    </tr>
    </tbody>
</table>

## Description

> When using the NumericTextBox with arrows enabled, I want to disable the down arrow programmatically at the component's minimum value to prevent looping to the maximum value. 

Reviewer 2: ```suggestion
In a NumericTextBox with enabled arrows, I want to disable the down arrow programmatically at the component's minimum value to prevent looping to the maximum value. 
```

This KB article answers the following questions:

- How can I disable the NumericTextBox increment and decrement buttons based on the value?
- How can I conditionally disable the NumericTextBox arrows when reaching specified value limits?

> ## Solution

Reviewer: We should explain very clearly that this KB works correctly only if `Min` and `Max` **do not match the default min and max values of the `Value` type**. Otherwise the user can loop from max to min and vice-versa with the arrow keys.

```razor
<style>
    .disable-increase .k-spinner-increase,
    .disable-decrease .k-spinner-decrease {
        pointer-events: none;
        opacity: 0.5;
    }
</style>

<TelerikNumericTextBox @bind-Value="@theValue"
                       Class="@numericClass"
                       Width="300px">
</TelerikNumericTextBox>

@code {
    private int theValue { get; set; } = int.MaxValue;
    private int minValue { get; set; } = int.MinValue;
    private int maxValue { get; set; } = int.MaxValue;

    private string numericClass => $"disable-arrows {(theValue == maxValue ? "disable-increase" : "")} {(theValue == minValue ? "disable-decrease" : "")}";
}
```

> To prevent the increase/decrease arrows of the NumericTextBox from being used when the numeric value reaches its minimum or maximum, apply a conditional CSS class to disable the buttons. 

Reviewer 2: ```suggestion
To prevent your end users from looping through the minimum and maximum values with the icnrease/decrease arrows of the NumericTextBox, apply a conditional CSS class to disable the buttons.
```

The example below demonstrates how to conditionally render CSS styles to disable the increase or decrease arrows based on the current value of the NumericTextBox.

````CSHTML
<style>
    .disable-increase .k-spinner-increase,
    .disable-decrease .k-spinner-decrease {
        pointer-events: none;
        opacity: 0.5;
    }
</style>

<TelerikNumericTextBox @bind-Value="@theValue"
                       Min="@minValue"
                       Max="@maxValue"
                       Class="@numericClass"
                       Width="300px">
</TelerikNumericTextBox>

@code {

>     private int theValue { get; set; } = 3;

Reviewer: Use capital letters for variables. See our Docs & Demos Code style.

```suggestion
    private int NumericValue { get; set; } = 3;
```

    private int minValue { get; set; } = 1;

>     private int maxValue { get; set; } = 10;

Reviewer: You can completely remove these variables to make the example simpler and shorter.

>     private string numericClass => $"disable-arrows {(theValue == maxValue ? "disable-increase" : "")} {(theValue == minValue ? "disable-decrease" : "")}";

Reviewer: What is the purpose of `disable-arrows` ?
